### PR TITLE
Mobile chart selection

### PR DESCRIPTION
**Pull request by dirkhh**
### Describe the pull request:

- [x] Functional change
- [x] New feature

### Pull request long description:

With all due respect to @bstoeger and #3162 - I think this is maybe slightly more reasonable

I gave up on doing this in the context drawer since Berthold thought this was a terrible idea, instead I took some liberty with his idea of a popup driven by a button, including actually showing the icons that he so carefully exported, styling the whole thing, placing it in the right spot in the layout, etc...

### CODE REVIEW

File: mobile-widgets/statsmanager.cpp [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
 
-StatsManager::StatsManager() : view(nullptr)
+StatsManager::StatsManager() : view(nullptr), charts(nullptr)
 {
-	// Test: show some random data. Let's see what happens.
-	state.var1Changed(2);
-	state.var2Changed(3);
-	state.binner2Changed(2);
+	// Default chart is dives per year
+	state.var1Changed(0);
```
Comment (bstoeger) [inline mobile-widgets/statsmanager.cpp]: If you want the default, you can remove these three lines.

**Review by bstoeger — APPROVED** [review]
Yes, looks much better.

There obviously is room for improvement (centring, resize the popup according to content), but it's a good start.

File: mobile-widgets/statsmanager.cpp (hunk 1 above)
Comment (dirkhh) [reply mobile-widgets/statsmanager.cpp]: ahh, didn't realize that. will do

**Comment by dirkhh** [comment]
> Yes, looks much better.
> 
> There obviously is room for improvement (centring, resize the popup according to content), but it's a good start.

It turns out that QML makes it hard to dynamically size the popup. Which is of course rather stupid, but, well, it's QML...
I can poke at this some more.
Right now I am working on fixing some other annoying issues with the QML UI - we are approaching the cross over point... roughly the same number of people have installed the recent iOS and Android updates as the total number of desktop users for 4.9.5 and higher... (admittedly that latter number could be under-counting with people who don't connect to our servers after an update). As a result I think it makes sense to pay more attention to things in the mobile UI that suck...

**Comment by dirkhh** [comment]
Ok, this is looking better. Please check my addition of that property - I couldn't make it work without adding that.

**Comment by bstoeger** [comment]
I believe the "correct" way to do this is to create a QML Model based on `QQMLInstanceModel`, which (of course!) is a private implementation detail of QML. Gee, thanks!

I would not another member variable, since this is redundant information and just risks getting inconsistent. Instead try something like
```
	Q_PROPERTY(int count READ rowCount NOTIFY countChanged)
```
(completely untested)

**Comment by dirkhh** [comment]
and how would that `countChanged()` signal fire? The same way as it fires today. Which means nothing changes from an __inconsistency__ point of view.
In reality this is really an implementation detail, but I'm happy to change accordingly.

**Comment by bstoeger** [comment]
Unless the code grows and we have to handle dynamic addition/removal of rows and forget to update count. Poof - `count != rowCount()`.

Of course, with a seasoned C programmer I would argue that it spares us 4 bytes of RAM. 😁

**Comment by dirkhh** [comment]
Please look at the modified implementation I just pushed

**Comment by bstoeger** [comment]
I think you forgot to commit, because nothing changed...?

**Comment by dirkhh** [comment]
Oh GitHub. Pushed again, now it seems to have accepted the force push. The interesting commit is 465a505

**Comment by bstoeger** [comment]
Sigh. I forgot that rowCount() takes a parameter. Still better than redundant state, IMHO.

**Comment by dirkhh** [comment]
> Sigh. I forgot that rowCount() takes a parameter. Still better than redundant state, IMHO.

Yeah, I see your point there. I didn't even realize this was possible.
I'm learning more from you every day, I guess :-)

Are you ok with merging this?

**Comment by bstoeger** [comment]
Sure, go ahead...

**Pull request merged**